tests/integration/helpers.py

```diff
@@ -372,3 +373,142 @@ def cluster_name(unit: Unit, model_name: str) -> str:
+    """
+    # get available unit id for the desidered application
+    units_ids = [
+        app_unit.name.split("/")[1]
+        for app_unit in ops_test.model.applications[application_name].units
+    ]
+    assert len(units_ids) > 0
+    unit_name = f"{application_name}/{units_ids[0]}"
+    raw_data = (await ops_test.juju("show-unit", unit_name))[1]
+    if not raw_data:
+        raise ValueError(f"no unit info could be grabbed for {unit_name}")
+    data = yaml.safe_load(raw_data)
+    # Filter the data based on the relation name.
+    relation_data = [v for v in data[unit_name]["relation-info"] if v["endpoint"] == relation_name]
+    if len(relation_data) == 0:
+        raise ValueError(
+            f"no relation data could be grabbed on relation with endpoint {relation_name}"
+        )
+
+    return relation_data
+
+
+def get_read_only_endpoint(relation_data: list) -> Set[str]:
+    """Returns the read-only-endpoints from the relation data.
+
+    Args:
+        relation_data: The dictionary that contains the info
+    Returns:
+        a set that contains the read-only-endpoints
+    """
+    related_units = relation_data[0]["related-units"]
+    roe = set()
+    for _, r_data in related_units.items():
+        assert "data" in r_data
+        data = r_data["data"]["data"]
+
+        try:
+            j_data = json.loads(data)
+            if "read-only-endpoints" in j_data:
+                read_only_endpoints = j_data["read-only-endpoints"]
+                if read_only_endpoints.strip() == "":
+                    continue
+                for ep in read_only_endpoints.split(","):
+                    roe.add(ep)
+        except json.JSONDecodeError:
+            raise ValueError("Relation data are not valid JSON.")
+
+    return roe
+
+
+def get_read_only_endpoint_hostnames(relation_data: list) -> List[str]:
+    """Returns the read-only-endpoint hostnames from the relation data.
+
+    Args:
+        relation_data: The dictionary that contains the info
+    Returns:
+        a set that contains the read-only-endpoint hostnames
+    """
+    roe = get_read_only_endpoint(relation_data)
+    roe_hostnames = []
+    for r in roe:
+        if ":" in r:
+            roe_hostnames.append(r.split(":")[0])
+        else:
+            raise ValueError("Malformed endpoint")
+    return roe_hostnames
+
+
+async def remove_leader_unit(ops_test: OpsTest, application_name: str):
+    """Removes the leader unit of a specified application.
+
+    Args:
+        ops_test: The ops test framework instance
+        application_name: The name of the application
+    """
+    leader_unit = None
+    for app_unit in ops_test.model.applications[application_name].units:
+        is_leader = await app_unit.is_leader_from_status()
+        if is_leader:
+            leader_unit = app_unit.name
+
+    units_to_destroy = [leader_unit]
+
+    for unit_to_destroy in units_to_destroy:
+        await ops_test.model.destroy_units(unit_to_destroy)
```

Comment:
why not just `await ops_test.model.destroy_units(leader_unit)`?

Reply:
Sure!